src/Nethermind/Nethermind.Serialization.Rlp/ReceiptArrayStorageDecoder.cs

```diff
@@ -0,0 +1,156 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using Nethermind.Core;
+using Nethermind.Core.Crypto;
+
+namespace Nethermind.Serialization.Rlp;
+
+public class ReceiptArrayStorageDecoder : IRlpStreamDecoder<TxReceipt[]>
+{
+    public static readonly ReceiptArrayStorageDecoder Instance = new();
+    private ReceiptStorageDecoder StorageDecoder = ReceiptStorageDecoder.Instance;
+    private CompactReceiptStorageDecoder CompactReceiptStorageDecoder = CompactReceiptStorageDecoder.Instance;
+
+    public const int CompactEncoding = 127;
```

Comment:
This won't ever conflict with proper RLP right? Do we need to make it explicit? Could we make it implicit by looking at RLP? Might be still better to make it explicit.

Reply:
Same trick as receipt type, should not conflict with RLP, first byte must be less than 128. Looking at RLP first makes things very complicated and error prone considering the many options available.